bundles/CoreBundle/src/Migrations/Version20240222143211.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Model\Dao\AbstractDao;
+
+/**
+ * Modifies foreign key constraints for specified tables, changing their names from ending in '_o_id' to '_id' and adding ON DELETE CASCADE.
+ * The down method reverses the naming back to '_o_id' while keeping ON DELETE CASCADE.
+ */
+final class Version20240222143211 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Dynamically rename _o_id foreign keys to _id.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('SET foreign_key_checks = 0');
+
+        foreach($schema->getTables() as $table) {
+            if($table->hasColumn("id")) {
+                $tableName = $table->getName();
+
+                if (
+                    str_starts_with($tableName, "object_brick_") ||
+                    str_starts_with($tableName, "object_classificationstore_") ||
+                    str_starts_with($tableName, "object_collection_") ||
+                    str_starts_with($tableName, "object_metadata_")
+                ) {
+                    $currentForeignKeyName = AbstractDao::getForeignKeyName($tableName, 'id');
+                    $foreignKeyWithOPrefix = preg_replace('/__id$/', '__o_id', $currentForeignKeyName);
+                    $foreignKeyWithoutOPrefix = preg_replace('/__o_id$/', '__id', $currentForeignKeyName);
```

Comment:
```suggestion
                    $currentForeignKeyName = AbstractDao::getForeignKeyName($tableName, 'id');
                    $foreignKeyWithOPrefix = AbstractDao::getForeignKeyName($tableName, 'o_id');
```
no need for `$foreignKeyWithoutOPrefix`
because `$currentForeignKeyName = AbstractDao::getForeignKeyName($tableName, 'id');` is already getting the one without `o_` prefix.
Also no need regex replace as it's a simple string concatenation.
See https://github.com/pimcore/pimcore/blob/68f198d32add713f2e4025b5dcb6414754133145/lib/Model/Dao/AbstractDao.php#L106-L108